### PR TITLE
feat: re-enable UNDO and REDO

### PR DESCRIPTION
**Pull request by gpuente**
## Description:

- re-enable UNDO and REDO

Screenshot 2024-07-22 at 4 43 21 PM

Resolves powerhouse-inc/document-model#75
Resolves powerhouse-inc/document-model#76

### CODE REVIEW

**Review by ryanwolhuter — APPROVED** [review]
nice!

File: src/document/reducer.ts [hunk 1]
```diff
@@ -83,6 +90,7 @@ function updateOperations<T extends Document>(
     document: T,
     action: Action | Operation,
     skip = 0,
+    reuseLastOperationIndex = false,
```
Comment (acaldas) [inline src/document/reducer.ts:93]: Please add a `@param` comment for the `skip` and `reuseLastOperationIndex`

File: src/document/reducer.ts [hunk 2]
```diff
@@ -140,8 +152,14 @@ export function updateDocument<T extends Document>(
     document: T,
     action: Action,
     skip = 0,
+    reuseLastOperationIndex = false,
```
Comment (acaldas) [inline src/document/reducer.ts:155]: Same here

File: src/document/reducer.ts [hunk 3]
```diff
@@ -314,29 +384,35 @@ export function baseReducer<T, A extends Action, L>(
...
+    let skipValue = skip || 0;
     let newDocument: Document<T, A, L> = { ...document };
-    // let clipboard = [...document.clipboard];
+    let reuseLastOperationIndex = false;
 
     const shouldProcessSkipOperation =
         !ignoreSkipOperations &&
         (skipValue > 0 || ('index' in _action && _action.skip > 0));
 
     // ignore undo redo for now
```
Comment (acaldas) [inline src/document/reducer.ts:396]: I guess this can be removed now :)

**Review by acaldas — APPROVED** [review]

File: src/document/reducer.ts (hunk 3 above)
Comment (gpuente) [reply src/document/reducer.ts]: fixed: [e66bcca](https://github.com/powerhouse-inc/document-model/pull/89/commits/e66bcca058fbef5dbb2543fc155ac580437e1a61)

File: src/document/reducer.ts (hunk 2 above)
Comment (gpuente) [reply src/document/reducer.ts:159]: fix: [e66bcca](https://github.com/powerhouse-inc/document-model/pull/89/commits/e66bcca058fbef5dbb2543fc155ac580437e1a61)

File: src/document/reducer.ts (hunk 1 above)
Comment (gpuente) [reply src/document/reducer.ts:95]: fix: [e66bcca](https://github.com/powerhouse-inc/document-model/pull/89/commits/e66bcca058fbef5dbb2543fc155ac580437e1a61)

**Pull request merged**